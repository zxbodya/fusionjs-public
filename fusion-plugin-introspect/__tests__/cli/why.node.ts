// @flow
import {mockFs} from './mockFs.js';
import {why} from '../../src/cli/why.js';

test('why', async () => {
  mockFs({
    version: '0.0.0',
    server: [
      {
        timestamp: 0,
        dependencies: [
          {
            name: 'A',
            type: 'both',
            sources: [{type: 'register', source: 'foo/bar:1:2'}],
            dependencies: ['B'],
          },
          {
            name: 'B',
            type: 'both',
            sources: [{type: 'register', source: 'foo/bar:3:4'}],
            dependencies: ['C'],
          },
          {
            name: 'C',
            type: 'both',
            sources: [{type: 'register', source: 'foo/bar:5:6'}],
            dependencies: [],
          },
        ],
      },
    ],
  });
  const data = await why('B');
  expect(data).toMatchSnapshot();
});
