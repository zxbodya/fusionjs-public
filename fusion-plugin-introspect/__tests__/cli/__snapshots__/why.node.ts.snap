// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`why 1`] = `
"B is registered in:
  foo/bar:3:4

B is used by:
└─ A

B depends on:
└─ C
"
`;
